File: SDKs/amazonaws.com/waf/2015-08-24/java/lib/src/main/java/openapisdk/models/shared/CreateWebAclRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

public class CreateWebAclRequest {
    @JsonProperty("ChangeToken")
    public String changeToken;
    public CreateWebAclRequest withChangeToken(String changeToken) {
        this.changeToken = changeToken;
        return this;
    }
    @JsonProperty("DefaultAction")
    public WafAction defaultAction;
    public CreateWebAclRequest withDefaultAction(WafAction defaultAction) {
        this.defaultAction = defaultAction;
        return this;
    }
    @JsonProperty("MetricName")
    public String metricName;
    public CreateWebAclRequest withMetricName(String metricName) {
        this.metricName = metricName;
        return this;
    }
    @JsonProperty("Name")
    public String name;
    public CreateWebAclRequest withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Tags")
    public Tag[] tags;
    public CreateWebAclRequest withTags(Tag[] tags) {
        this.tags = tags;
        return this;
    }
}

File: SDKs/amazonaws.com/waf/2015-08-24/python/src/sdk/models/shared/createwebaclrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import wafaction as shared_wafaction
from ..shared import tag as shared_tag


@dataclass_json
@dataclasses.dataclass
class CreateWebACLRequest:
    change_token: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ChangeToken') }})
    default_action: shared_wafaction.WafAction = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('DefaultAction') }})
    metric_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('MetricName') }})
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    tags: Optional[list[shared_tag.Tag]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    
